scripts/db_migrator.py

```diff
@@ -96,6 +100,89 @@ def migrate_interface_table(self):
+
+        # Get platform info and hwsku from config_db.json
+        config_db_file_path = "/etc/sonic/config_db.json"
+        if os.path.exists(config_db_file_path):
+            try:
+                with open(config_db_file_path) as config_db_file:
+                    config_db = json.load(config_db_file)
+                    device_data = config_db['DEVICE_METADATA']
+                    hwsku = device_data['localhost']['hwsku']
+                    platform = device_data['localhost']['platform']
+                    buffer_pool_conf = config_db['BUFFER_POOL']
+            except IOError:
+                log_error("failed to open config_db.json file, skip migration")
+                return False
+        else:
+            # Try to get related info from DB(in warm reboot case DB will be retored)
+            buffer_pool_conf = self.configDB.get_table('BUFFER_POOL')
+            device_data = self.configDB.get_table('DEVICE_METADATA')
+            if 'localhost' in device_data.keys():
+                hwsku = device_data['localhost']['hwsku']
+                platform = device_data['localhost']['platform']
+            else:
+                log_error("Trying to get data from DB but it's empty, skip migration")
+                return False
+
+        # Get current buffer pool configuration, only migrate configration which 
+        # with default values, if it's not default, leave it as is.
+        pool_size_in_db_list = []
+        pools_in_db = buffer_pool_conf.keys()
+
+        # Buffer pool numbers is different with default, don't need migrate
+        if len(pools_in_db) != len(buffer_pools):
+            return True
+
+        # If some buffer pool is not default ones, don't need migrate
+        for buffer_pool in buffer_pools:
+            if buffer_pool not in pools_in_db:
+                return True
+            pool_size_in_db_list.append({buffer_pool: buffer_pool_conf[buffer_pool]['size']})
+        
+        # To check if the buffer pool size is equal to old default values
+        if pool_size_in_db_list == spc1_t0_default_value or pool_size_in_db_list == spc1_t1_default_value \
+            or pool_size_in_db_list == spc2_t0_default_value or pool_size_in_db_list == spc2_t1_default_value:
+            # Generate buffer configuration from latest template
+            tmp_json_file = "/tmp/mlnx_buffers_all.json"
+            device_buffer_template_path = os.path.join("/usr/share/sonic/device/", platform, hwsku, 'buffers.json.j2')
+            command = "sonic-cfggen -d -t {} >{}".format(device_buffer_template_path, tmp_json_file)
+            try:
+                subprocess.check_call(command, shell=True)
+            except:
+                log_error("failed to generate new mlnx buffer configuration, skip migration")
+                return False
```

Comment:
Can you make code robust enough so that failure is not happening?

Having a failure in migration path would leave config_db not migrated, and likely will cause orchagent to crash in the future. right?

Reply:
deprecate the original way to generate a tmp buffer config json file with new configurations, instead, use predefined new buffer pool configuration dict,  this will reduce the possible fail case during handling the tmp files.